Experimental API

druzhkov-paul opened the pull request.

AlexKoff88 commented on init_venv.sh:
```diff
@@ -0,0 +1,81 @@
+# Create virtual environment
+virtualenv ${venv_dir} -p python3 --prompt="(detection)"
+
+path_openvino_vars="${INTEL_OPENVINO_DIR:-/opt/intel/openvino_2021}/bin/setupvars.sh"
```
I suggest migrating to `pip install openvino openvino-dev`. Benchmark_app and AccuracyChecker are already on board.

eugene123tw committed:
use mmdatacpu and mmdataparallel

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/add_template_for_pot_nncf_tests

Commit:
make nncf optional for export

druzhkov-paul committed:
Merge pull request #203 from openvinotoolkit/eugene/bump_mmcv_version

bump mmcv version to 1.3.14 and add support for VFNet CPU

LeonidBeynenson committed:
Remove unrequired ModelEntity fields in test_ote_training.py

Commit:
add environment info to logs

LeonidBeynenson committed:
Debug insertions into EvalHook

LeonidBeynenson committed:
Fix small issue with zero metric in AccuracyAwareRunner

Commit:
extend constraints

LeonidBeynenson committed:
Revert "Debug insertions into EvalHook"

This reverts commit a4b49f04abfe7341c4a27ac8fd4b420663479f3c.

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/add_template_for_pot_nncf_tests

Commit:
fix torch version

Commit:
clean up

eugene123tw committed:
add adaptive repeat feature

Commit:
verify gpu and cpu environments

Commit:
update sdk hash

Commit:
fix google-auth version for e2e

Commit:
remove set_values_as_default

Commit:
use syntetic dataset in sample

Commit:
fix test

LeonidBeynenson committed:
Decorative change tests/test_ote_training.py

Co-authored-by: Pavel Druzhkov <druzhkov.paul@gmail.com>

LeonidBeynenson committed:
Add validation of reallife tests results in test_ote_training.py

LeonidBeynenson committed:
Merge branch 'lb/add_template_for_pot_nncf_tests' into lb/reallife_training_tests

Commit:
Update configs for September candidates

Commit:
Update readme metrics for ATSS since more accurate model were used as started snapshot

druzhkov-paul committed:
Merge pull request #211 from openvinotoolkit/eugene/adaptive_repeat_ds

Adaptive dataset repetition

Commit:
skip irrelevant tests

Commit:
SyncBN -> BN to enable CPU inference

LeonidBeynenson committed:
Decorative changes in test_ote_training.py

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/add_template_for_pot_nncf_tests

LeonidBeynenson committed:
Fix small bug in the last decorative change in test_ote_training.py

LeonidBeynenson committed:
Add comments and docstrings to new fixtures in test_ote_training.py

druzhkov-paul committed:
Merge pull request #212 from openvinotoolkit/gz/custom_OD_candidates

Config updates for latest custom OD candidates

druzhkov-paul committed:
unset debug log level

druzhkov-paul committed:
Merge pull request #205 from openvinotoolkit/lb/add_template_for_pot_nncf_tests

Add pot and nncf tests

Commit:
fix type annotation

Commit:
Merge branch 'dp/fix_tests' into dp/log_env_2

Commit:
fix importlib-metadata version

druzhkov-paul committed:
Merge pull request #214 from openvinotoolkit/dp/fix_tests

Fix type annotation in tests

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/reallife_training_tests

druzhkov-paul committed:
Merge pull request #210 from openvinotoolkit/dp/log_env

Environment logging

druzhkov-paul committed:
Merge pull request #209 from openvinotoolkit/dp/fix_no_nncf_export

Fix export when no NNCF installed

Commit:
use global constraints file in init_venv script

Commit:
set not_improved status

druzhkov-paul committed:
Merge pull request #217 from openvinotoolkit/dp/not_improved_status

Properly set NOT_IMPROVED status

Commit:
Revert "Merge pull request #203 from openvinotoolkit/eugene/bump_mmcv_version"

This reverts commit d7f0733d57fcfb2dcbb864b2fcbf740c9914e10c, reversing
changes made to 673bbd73cd6368c81de3e3e17c2153e054cd79c9.

Commit:
update metrics to use updated API

druzhkov-paul committed:
Merge pull request #218 from openvinotoolkit/dp/revert_mmcv_update

Revert #203

druzhkov-paul committed:
Merge pull request #216 from openvinotoolkit/dp/global_constraints

Global constraints for init_venv

Commit:
Revert #218. Repeat #203.

This reverts commit 960c2d34998eb74e925703fd86ef78184dd60088, reversing
changes made to 5fe1c3c07cf96e942502aac76fb253cbab81c691.

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/reallife_training_tests

druzhkov-paul committed:
Merge pull request #219 from openvinotoolkit/dp/metric_groups

Update metrics to use new API

LeonidBeynenson committed:
Small update test_ote_training.py

Co-authored-by: Pavel Druzhkov <druzhkov.paul@gmail.com>

LeonidBeynenson committed:
Remove using `num_checkpoints` parameter from test_ote_training.py

LeonidBeynenson committed:
Decorative change in test_ote_training.py

Co-authored-by: Pavel Druzhkov <druzhkov.paul@gmail.com>

LeonidBeynenson committed:
Add validation step to test_ote_training.py for stages that were run from dependency chain earlier

It is possible that a stage is run from a dependency chain (for some
other stage) and later it is run not from dependency chain.
In this case the stage's action will be run in the first run only, but
validation will be run each time when the stage is run not from a
dependency chain.

Ilya-Krylov committed:
added cu102

druzhkov-paul committed:
Merge pull request #222 from openvinotoolkit/ik/cu102

added cu102

e-ddykim committed:
enable hpo in mmdetection

e-ddykim committed:
HPO search space is updated

e-ddykim committed:
code to call update_progress_callback is updated

e-ddykim committed:
updated to pass a score as a named argument

Co-authored-by: Pavel Druzhkov <druzhkov.paul@gmail.com>

Commit:
add deploy method

e-ddykim committed:
explicit removal of the 1st item in a list

druzhkov-paul committed:
Merge pull request #223 from openvinotoolkit/hpo_bnw

enable HPO for OTE_cli (brave_new_world)

Commit:
dump features

Commit:
update sdk sha

Commit:
clean up

LeonidBeynenson committed:
Move getting expecting metric to callback (in test_ote_training.py)

It is required to avoid attempts to get expected validation metrics for
tests that do not require validation at all.

Commit:
add model stats (gigaflops and size)

Commit:
fix infer output

LeonidBeynenson committed:
Give possibility to point complicated acceptable ranges of expected metrics in test_ote_training.py

druzhkov-paul committed:
Merge pull request #225 from openvinotoolkit/dp/dump_features

Dump features during the inference

druzhkov-paul committed:
Merge pull request #226 from openvinotoolkit/dp/model_stats

Add model stats to templates

Commit:
Merge branch 'dp/nous_sdk_api' into dp/update_mmcv_to_3_14

Commit:
add config

Commit:
update package content

Commit:
enable adaptive confidence threshold selection at the training stage only

Commit:
fix docs

Commit:
fix nncf task

Commit:
disable tensorboard logging by default

Commit:
remove checks for model improvement

Commit:
fix best checkpoint loading

Commit:
workaround for nncf progress tracking

Commit:
add model template ids and change names

druzhkov-paul committed:
Merge pull request #228 from openvinotoolkit/dp/adaptive_confidence_threshold

Adaptive confidence threshold

druzhkov-paul committed:
Merge pull request #229 from openvinotoolkit/dp/template_ids

Add model template IDs

Commit:
integrate models from ModelAPI

Commit:
Merge remote-tracking branch 'origin/dp/nous_sdk_api' into ak/model_in_ote_api

Commit:
add optional location for OTE SD

Commit:
fix progress callbacks in tests

Commit:
remove gen1 and gen2 templates

Commit:
refactor confidence threshold handling

druzhkov-paul committed:
fix conditions in if-else clause

Co-authored-by: Leonid Beynenson <leonid.beynenson@intel.com>

druzhkov-paul committed:
Merge pull request #230 from openvinotoolkit/dp/ote_sdk_path_envvar

Optional SDK location

druzhkov-paul committed:
Merge pull request #231 from openvinotoolkit/dp/progress_tracking_tests

Fix progress callbacks in tests

druzhkov-paul committed:
Merge pull request #232 from openvinotoolkit/dp/remove_old_templates

Remove Gen1 and Gen2 templates

druzhkov-paul committed:
Merge pull request #233 from openvinotoolkit/dp/refactor_confidence_threshold

Refactor confidence threshold handling

Commit:
add flag for new models

Commit:
Merge branch 'dp/nous_sdk_api' into dp/update_mmcv_to_3_14

druzhkov-paul committed:
Merge pull request #220 from openvinotoolkit/dp/update_mmcv_to_3_14

Revert #218. Repeat #203.

Commit:
add possibility to use default parameters from ModelAPI

LeonidBeynenson committed:
Make evaluation on TESTING subset in test_ote_training.py

Commit:
add logs for profiling

Commit:
Merge remote-tracking branch 'origin/dp/nous_sdk_api' into ak/model_in_ote_api

Commit:
increase default number of pre-processing workers

eugene123tw committed:
add adaptive repeat dataset on VFNet

add adaptive repeat on SSD

Commit:
update inference parameters

druzhkov-paul committed:
Merge pull request #236 from openvinotoolkit/eugene/patch_vfnet

VFNet with adaptive repeat dataset

LeonidBeynenson committed:
Add one more exception handling in test_ote_training.py

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/reallife_training_tests

Commit:
revert read-only parameter and change docs

Commit:
use mmdet logger

Commit:
do not disable the text logger hook

Commit:
increase training log verbosity

druzhkov-paul committed:
Merge pull request #237 from openvinotoolkit/dp/profiling_logs

Profiling logs

Commit:
Add clustering for SSD model

AlexanderDokuchaev committed:
Set correct model format - pytorch

AlexanderDokuchaev committed:
Fix undefined variable

Commit:
remove num_checkpoints from test setup

AlexanderDokuchaev committed:
Add check model format

Commit:
use develop HEAD for sdk version

AlexanderDokuchaev committed:
sdk

Commit:
Fixes from comments

druzhkov-paul committed:
Merge pull request #239 from openvinotoolkit/ad/fix_optimized_model_format

Fix model format for optimized model

druzhkov-paul committed:
Merge pull request #240 from openvinotoolkit/dp/sdk_develop

Use develop HEAD for SDK version

druzhkov-paul committed:
Merge pull request #238 from openvinotoolkit/dp/fix_hyperparams_in_tests

Remove num_checkpoints from test setup

Commit:
Add loading stats from snapshot in nncf task

Commit:
Move functions to the data_utils

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/reallife_training_tests

Commit:
fix annotation scene assignment in dataset item

Commit:
Merge branch 'dp/nous_sdk_api' into dp/fix_annotation_scene_assignment

Commit:
Fixes from comments and tests fixes

LeonidBeynenson committed:
Improve handling wrong config errors in test_ote_training.py

Also add comments and todos.

druzhkov-paul committed:
Merge pull request #243 from openvinotoolkit/dp/fix_annotation_scene_assignment

Fix annotation scene assignment in dataset item

Commit:
update inferencer

Commit:
add folder for wrappers

Commit:
Merge remote-tracking branch 'origin/dp/nous_sdk_api' into ak/model_in_ote_api

Commit:
Commit test file

Commit:
based on new model api

LeonidBeynenson committed:
Replace obsolete Polygon3 with shapely.geometry.Polygon

Note that the change was not debugged properly.

LeonidBeynenson committed:
Fix small bug

druzhkov-paul committed:
Merge pull request #215 from openvinotoolkit/lb/reallife_training_tests

Reallife training tests

LeonidBeynenson committed:
Remove unrequired comment

The comment is not required since we checked that updated code works
with shapely.geometry.Polygon properly.

Commit:
cast detector output to Python float

druzhkov-paul committed:
Merge pull request #241 from openvinotoolkit/gz/anchor_clustering_ssd

Clustering for SSD

druzhkov-paul committed:
Merge pull request #244 from openvinotoolkit/lb/replace_Polygon_with_shapely_polygon

Replace polygon with shapely polygon

druzhkov-paul committed:
Merge pull request #245 from openvinotoolkit/dp/numpy_float_type

Fix output array type in OpenVINO inferencer

Commit:
get current lr from optimizer (instead of initial lr)

Commit:
connect metadata to the model

Ilya-Krylov committed:
temp

Commit:
fix iteration indices in learning curves

AlexanderDokuchaev committed:
fix attributes of optimized model

Commit:
remove nan values

Commit:
pass `lr` from optimizer params to the 'initial_lr' if it's not saved

LeonidBeynenson committed:
Move dependency inside tesr actions in test_ote_training.py

AlexanderDokuchaev committed:
fix

LeonidBeynenson committed:
Make a hotfix to show test_ote_training.py test results in the dashboard properly

LeonidBeynenson committed:
Add in test_ote_training.py one more field for dashboard

druzhkov-paul committed:
Merge pull request #249 from openvinotoolkit/dp/metadata_model

Connect metadata to the model

AlexanderDokuchaev committed:
-

druzhkov-paul committed:
Merge pull request #248 from openvinotoolkit/dp/learning_curves

Fix iteration indices in learning curves

Commit:
relax upper bound for expected metric value

AlexanderDokuchaev committed:
tests

druzhkov-paul committed:
Merge pull request #252 from openvinotoolkit/dp/relax_metric_range

Relax upper bound for expected metric value

druzhkov-paul committed:
Merge pull request #251 from openvinotoolkit/lb/hotfix_data_collector_fields_for_dashboard

Hotfix data collector fields for dashboard

druzhkov-paul committed:
Merge pull request #250 from openvinotoolkit/ad/fix_optimization_type

Fix attributes of optimized model

Ilya-Krylov committed:
save using torch.save

Ilya-Krylov committed:
Update train_task.py

Ilya-Krylov committed:
Update train_task.py

Ilya-Krylov committed:
Merge pull request #247 from openvinotoolkit/ik/label_schema_in_snapshot

Ik/label schema in snapshot

LeonidBeynenson committed:
Move generating params for test actions to factories in test_ote_training.py

Commit:
add base task capabilities to model templates

Commit:
fix e2e test

Commit:
replace `labels` to `label_schema` in tests to make them pass

Commit:
fix e2e tests

druzhkov-paul committed:
Merge pull request #253 from openvinotoolkit/dp/template_capabilities

Add base task capabilities to model templates

LeonidBeynenson committed:
Use generated class OTEIntegrationTestCase in test_ote_training.py

Ilya-Krylov committed:
speed up infer for very small dataset

Ilya-Krylov committed:
serialize

Ilya-Krylov committed:
Merge branch 'dp/nous_sdk_api' into ik/dict_serialized_label_schema

Ilya-Krylov committed:
Merge pull request #255 from openvinotoolkit/ik/speed_up_infer_for_1_image

speed up infer for very small dataset

LeonidBeynenson committed:
Extract test helper class from the main test class in test_ote_training.py

LeonidBeynenson committed:
Split test_ote_training.py into several files

druzhkov-paul committed:
Merge pull request #256 from openvinotoolkit/ik/dict_serialized_label_schema

dict serialized label schema

LeonidBeynenson committed:
Clean up imports in ote_training_test* files and test_ote_training.py

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/refactor_reallife_tests_to_move

Since in the branch `lb/refactor_reallife_tests_to_move`
the file `test_ote_training.py` was splitted into several files, the
changes made in the branch `dp/nous_sdk_api` in the file
`test_ote_training.py` were splitted into the files
`ote_training_tests_actions.py` and `test_ote_training.py`

LeonidBeynenson committed:
Switch to using algo backend test helpers from OTE SDK

Ilya-Krylov committed:
store label_schema separately from checkpoint

Commit:
update openvino_task and add deploy

Commit:
Merge branch 'dp/nous_sdk_api' into ak/model_in_ote_api

Commit:
Merge pull request #246 from openvinotoolkit/gz/fix/pass_lr_after_resuming_ckpt

Fix LR value after resuming training with ReduceOnPlateau optimizer

Commit:
add README to deployable package

LeonidBeynenson committed:
Move some fixtures and common pytest code to OTE SDK test helpers

LeonidBeynenson committed:
Finishing moving OTE test helpers to OTE SDK

evgeniya-egupova committed:
Shuffle pot data subset for a better quantization

Commit:
update requirements and tests

LeonidBeynenson committed:
Delete e2e_test_system.py file since it was moved to OTE SDK test helpers

eugene123tw committed:
ource /home/yuchunli/git/algo_mmdet/venv/bin/activate    This is a combination of 4 commits.

Rename template.yaml to template_experimental.yaml

fix test_reading_gen3_vfnet

fix hard coded template in setup_configurable_parameters

Commit:
chnage content of package

Commit:
remove some optional dependencies from the list

Ilya-Krylov committed:
tests fix

druzhkov-paul committed:
Merge pull request #262 from openvinotoolkit/dp/remove_bad_dependencies

Remove some optional dependencies

Commit:
update hyper_parameters for nncf_task

Commit:
update type of confidence_threshold

Commit:
Merge remote-tracking branch 'origin/dp/nous_sdk_api' into ak/model_in_ote_api

druzhkov-paul committed:
Merge pull request #261 from openvinotoolkit/eugene/rename_unused_tl

Rename template.yaml to template_experimental.yaml

druzhkov-paul committed:
Merge pull request #258 from openvinotoolkit/ik/serialize_label_schema_separately

store label_schema separately from checkpoint

Commit:
Merge remote-tracking branch 'origin/dp/nous_sdk_api' into ak/model_in_ote_api

LeonidBeynenson committed:
Change model names -- now OTE SDK test helpers read model names from template files

Now OTE SDK test helpers read model names from the field "model_template_id"
of model template files.

Also errors handling is improved a bit.

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/moving_reallive_training_tests

Commit:
update requirements for mmdetection

Commit:
update reqs

LeonidBeynenson committed:
Change test helpers imports from OTE SDK, since they were moved

LeonidBeynenson committed:
Make decorative changes w.r.t. PR comments

Commit:
useful microchanges

druzhkov-paul committed:
Merge pull request #264 from openvinotoolkit/ak/update_requirements

OpenVINO 2021.4.2

Commit:
Merge remote-tracking branch 'origin/dp/nous_sdk_api' into ak/model_in_ote_api

Commit:
return params structure

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/moving_reallive_training_tests

druzhkov-paul committed:
Merge pull request #227 from openvinotoolkit/ak/model_in_ote_api

Deployment via ModelAPI

LeonidBeynenson committed:
Merge branch 'dp/nous_sdk_api' into lb/moving_reallive_training_tests

SemyonBevzuk committed:
Add get_nms_from_type

druzhkov-paul committed:
Merge pull request #260 from openvinotoolkit/lb/moving_reallive_training_tests

Moving reallive training tests to OTE SDK test helpers

AlexanderDokuchaev committed:
Use percents for max_accuracy_drop

AlexanderDokuchaev committed:
Add prepare_for_export for nncf model, save anchors

AlexanderDokuchaev committed:
update nncf

Commit:
update visualization

druzhkov-paul committed:
Merge pull request #259 from evgeniya-egupova/pot_shuffle_dataset

Shuffle pot data subset for a better quantization

AlexanderDokuchaev committed:
Revert "update nncf"

This reverts commit bf15930c964fdeeafe953738ca1c61ac8dd2538e.

Commit:
Merge branch 'ote' into dp/nous_sdk_api

druzhkov-paul committed:
Merge pull request #268 from openvinotoolkit/ad/nncf_max_accuracy

Use percent for max_accuracy_drop

druzhkov-paul committed:
Merge pull request #270 from openvinotoolkit/dp/merge_ote_branch

Merge ote branch

Commit:
filter gt annotation by domain and labels

Commit:
fix tests

SemyonBevzuk committed:
Fix

SemyonBevzuk committed:
Fix analyze_logs.py

Ilya-Krylov committed:
Merge pull request #269 from openvinotoolkit/ak/update_visualization

Update visualization for OV inferencer

Ilya-Krylov committed:
Merge branch 'dp/nous_sdk_api' into dp/filter_gt_annotation

Ilya-Krylov committed:
Merge pull request #274 from openvinotoolkit/dp/filter_gt_annotation

Filter GT annotation by domain and labels

Ilya-Krylov committed:
Merge pull request #272 from openvinotoolkit/sb/batched_nms_eval

Fix problems with eval

Ilya-Krylov committed:
export via run shell=False

Ilya-Krylov committed:
fix

Ilya-Krylov committed:
Merge pull request #279 from openvinotoolkit/ik/export_via_run_shell_false

export via run shell=False

Commit:
OTEDetection -> MMDetection in docs and setup.py

Commit:
remove lxml from requirements as it isn't used

Commit:
replace relative paths to absolute paths in setup.py

Commit:
replace import xml.etree to defusedxml in datasets implementation

Commit:
update networkx version

Ilya-Krylov committed:
Merge pull request #282 from openvinotoolkit/gz/ote_sdl

Update docs and code according to the SDL requirements
